nmpolicy/internal/resolver/path.go

```diff
@@ -0,0 +1,83 @@
+package resolver
+
+import (
+	"fmt"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/ast"
+)
+
+func applyFuncOnPath(inputState interface{}, path []ast.Node, filterFn func(map[string]interface{}, string) (interface{}, error), shouldFilterSlice bool) (interface{}, error) {
+	if len(path) == 0 {
+		return inputState, nil
+	}
+	toFilterMap, isMap := inputState.(map[string]interface{})
+	if len(path) == 1 && isMap {
+		if filterFn != nil {
+			key := *path[0].Identity
+			outputState, err := filterFn(toFilterMap, key)
+			if err != nil {
+				return nil, err
+			}
+			return outputState, nil
+		}
+		return inputState, nil
```

Comment:
Let's return an error if we are at the end of recursion and it's not a map, so we have a place that enters only at the end of the recursion.

```golang
if len(path) == 1 {
   if !isMap {
      return nil, pathError....
  }
  ...
```